substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1749,6 +1768,10 @@ pub mod pallet {
 					if let Some(ref mut nom) = maybe_nom {
 						if let Some(pos) = nom.targets.iter().position(|v| v == stash) {
 							nom.targets.swap_remove(pos);
+
+							// update nominations and trickle down to target list score.
+							Self::do_add_nominator(&nom_stash, nom.clone());
```

Comment:
calling a function that inserts into `Nominators` inside a `Nominators::mutate` callback feels scary.